include/circt/Dialect/FIRRTL/FIRRTLAnnotations.h

```diff
@@ -82,6 +82,12 @@ class AnnotationSet {
   /// false otherwise.
   bool applyToOperation(Operation *op) const;
 
+  /// Store the annotations in this set in an operation's `annotations`
+  /// attribute, overwriting any existing annotations. Removes the `annotations`
+  /// attribute if the set is empty. Returns true if the operation was modified,
+  /// false otherwise.
+  bool applyToOperation(Operation &op) const { return applyToOperation(&op); }
```

Comment:
Just curious, but is this normal to have the same method defined which takes both a pointer to `A` and a reference to `A` or is it better to have just one?

Reply:
I *think* its fairly normal to add them as you need them.  I was thinking this one makes sense since the block iterators return Operation references.  I guess since I don't really know, I'll just drop this part from the change and use the existing one.

Reply:
Nah, I'm good to have it in! It seems usable, I just don't know what the official guidance on API design in C++ is here. If it seems natural to you, go for it.